playbooks/ansible/roles/ctdb.setup/tasks/main.yml

```diff
@@ -32,14 +32,22 @@
 - name: Update selinux contexts
   command: restorecon -R "{{ config.paths.ctdb.var_lib }}/lock"
 
-- name: Add ctdb rules to firewalld
-  firewalld:
-    service: ctdb
-    permanent: yes
-    state: enabled
-
-- name: Reload firewall
-  command: firewall-cmd --reload
+- name: Check firewall
+  command: firewall-cmd --state
+  failed_when: false
+  register: fwd_state
```

Comment:
Yes. This has been necessary because firewall configuration for GPFS is not trivial for each machine type and I preferred to implement it in a future patch, so GPFS disables the firewall. However, trying to configure the firewall when it's disabled, causes a failure, so I only do that if the firewall is actually enabled.